### PR TITLE
feat: improve UX for Typescript app codegen

### PR DESCRIPTION
**Pull request by Niraj-Kamdar**
Update App Template: https://github.com/polywrap/cli/pull/1855

### CODE REVIEW

**Review by krisbitney — COMMENTED** [review]
This looks good but I think we could reduce the number of getters for each property. I'm guessing you thought a lot about this so maybe I'm missing something.

Since the getter methods are abstract, the user will always implement them. Since they will always be implemented, I think it would make more sense to just make the properties themselves abstract and public, or to remove the properties altogether and just use the methods.

For example,

```ts
/ is this._defaultUri redundant?
private _getUri(uri?: string): string {
  return uri || this._defaultUri || this._getDefaultUri() || "testimport.uri.eth";
}
  // can we do this?
private _getUri(uri?: string): string {
  return uri || this._getDefaultUri() || "testimport.uri.eth";
}
  ```

**Review by krisbitney — APPROVED** [review]

File: implementations/app-typescript/src/__tests__/cases/000-sanity/output/types.ts [hunk 1]
```diff
@@ -183,42 +184,102 @@ export interface TestImport_Module_Args_returnsArrayOfEnums {
 }
 
 /* URI: "testimport.uri.eth" */
-export const TestImport_Module = {
-  importedMethod: async (
+export abstract class BaseTestImport {
```
Comment (dOrgJelli) [inline implementations/app-typescript/src/__tests__/cases/000-sanity/output/types.ts:187]: This type isn't being used anywhere, is this what the user imports into their code? If so I don't think that's what we want, instead we'd want `import { TestImport } from ...`

**Review by dOrgJelli — CHANGES_REQUESTED** [review]
This PR should point to a new PR in the CLI for the updated template project. This way we know what the e2e looks like.

**Comment by dOrgJelli** [comment]
Created a PR here @Niraj-Kamdar https://github.com/polywrap/cli/pull/1855

**Review by dOrgJelli — APPROVED** [review]
:fire:

**Pull request merged**